config/provider/upgrade/config.go

```diff
@@ -0,0 +1,11 @@
+package providerupgradeconfig
```

Comment:
Can we move our actual API types here to an `apis/` top-level directory? I think we should use `config/` for keeping the YAML that uses these API types, but the API types should be kept elsewhere :+1:

Reply:
Yes, that will be much better. I guess we can keep the directory structure as similar to that of crossplane (`apis/pkg/v1/provider.go`) and `provider.go` will be part of package `v1` which will contain code that resides in `config.go` right now. Also, I guess we can add `getConfiguration()` method here.
Also, will it be better to change the path for `config.yml` from `config/provider/upgrade` to `conifg/provider` since we might be using the same config file for other provider-related tests in the future?